scripts/views/mapView.js

```diff
@@ -173,6 +187,42 @@
     }
   };
 
+  // If no events of a given type are present in the snapshot, hide its associated legend li
+  mapView.hideAbsentEvents = function hideAbsentEvents(){
+    for(var key in legendIcons){
+      if(legendIcons[key] === false){
+        $(`.${key}-parent`).addClass('hidden');
```

Comment:
I might reverse this. Start with them hidden and removeClass('hidden) if the icon is true, that way the user doesn't see things disappearing from the legend. Also in JavaScript you can just do truthy checks: `if (legendIcon[key])` or `if(!legendIcon[key])`